src/shader_recompiler/backend/spirv/emit_spirv_image.cpp

```diff
@@ -157,8 +157,11 @@ Id EmitImageFetch(EmitContext& ctx, IR::Inst* inst, u32 handle, Id coords, const
     ImageOperands operands;
     operands.AddOffset(ctx, offset);
     operands.Add(spv::ImageOperandsMask::Lod, lod);
-    return ctx.OpBitcast(
-        ctx.F32[4], ctx.OpImageFetch(result_type, image, coords, operands.mask, operands.operands));
+    const Id texel =
+        texture.is_storage
+            ? ctx.OpImageRead(result_type, image, coords, operands.mask, operands.operands)
```

Comment:
I think you need casts here that match the return type in opcodes.inc like it did before
i.e. vec4 for fetch, uvec4 for read

Reply:
didnt see the next line. You do f32 for fetch but need to do u32 when its read, otherwise spirv-val errors

Reply:
Have you encountered any actual validation errors there? Because to me, it looks okay:
```
        %432 = OpCompositeConstruct %u32vec2_id %u32_id_0 %u32_id_0
        %433 = OpLoad %53 %cs_img12_18
        %434 = OpImageRead %f32vec4_id %433 %432 None
        %435 = OpBitcast %f32vec4_id %434
```